code/game/machinery/vending.dm

```diff
@@ -3706,3 +3706,52 @@ var/global/num_vending_terminals = 1
 		return pick(grey_slogans)
 	else
 		return pick(product_slogans)
+
+/obj/machinery/vending/lotto
+	name = "\improper Lotto Tickets"
+	desc = "Wall-mounted vending machine which dispenses scratch-off lottery tickets. Winners can be cashed at Cargo."
+	product_slogans = list(
+		"Feeling lucky?",
+		"Money won is twice as sweet as money earned.",
+		"The greatest risk is not taking one."
+	)
+	product_ads = list(
+		"Quit while you’re ahead. All the best gamblers do.",
+		"If there weren’t luck involved, I would win every time.",
+		"Better an ounce of luck than a pound of gold.",
+		"Behind bad luck comes good luck."
+	)
+	vend_reply = "Good luck!"
+	icon_state = "Lotto"
+	icon_vend = "Lotto-vend"
+	products = list(
+		/obj/item/toy/lotto_ticket/gold_rush = 20,
+		/obj/item/toy/lotto_ticket/diamond_hands = 20,
+		/obj/item/toy/lotto_ticket/phazon_fortune = 20
+		)
+	contraband = list(
+		/obj/item/toy/lotto_ticket/supermatter_surprise = 1
+		)
+	prices = list(
+		/obj/item/toy/lotto_ticket/gold_rush = 5,
+		/obj/item/toy/lotto_ticket/diamond_hands = 20,
+		/obj/item/toy/lotto_ticket/phazon_fortune = 50,
+		/obj/item/toy/lotto_ticket/supermatter_surprise = 100
+		)
+
+	pack = /obj/structure/vendomatpack/lotto
+
+/obj/machinery/vending/lotto/attackby(obj/item/I, user)
+	add_fingerprint(user)
+	if(istype(I, /obj/item/toy/lotto_ticket))
+		var/obj/item/toy/lotto_ticket/T = I
+		if(!T.iswinner)
+			playsound(src, "buzz-sigh", 50, 1)
+			for(var/mob/V in hearers(src))
+				V.visible_message("<b>[src]</b>'s monitor flashes, \"This ticket is not a winning ticket.\"")
```

Comment:
`visible_message` is meant to replace the for-loop, not be used within in. This will send a message to every mob as many times as there are mobs in sight.

Reply:
ok